lib/loader.ts

```diff
@@ -3,10 +3,11 @@ import { register, RegisterOptions } from "ts-node";
 import { TypeScriptCompileError } from "./typescript-compile-error";
 
 export function TypeScriptLoader(options?: RegisterOptions): Loader {
+  const tsNodeInstance = register({ ...options, compilerOptions: { module: "commonjs" } })
   return (path: string, content: string) => {
     try {
       // cosmiconfig requires the transpiled configuration to be CJS
-      register({ ...options, compilerOptions: { module: "commonjs" } }).compile(
+      tsNodeInstance.compile(
```

Comment:
Yes, been on PTO, will get then asap, thanks!

Reply:
I'm confused by the failing test "rethrows an error if it is not an instance of Error", what are we testing here?

Reply:
What could throw errors from the register, I feel like a misconfiguration of the loader should be handled differently from the errors throw from compile, as loader is going to fail for the developer setting up the loader rather than later based on a random config file issue.

Reply:
Makes sense for a major, while not an API change, it could impact current users negatively.